jsonschema-module-jackson/pom.xml

```diff
@@ -34,6 +35,17 @@
             <artifactId>jackson-annotations</artifactId>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>org.jetbrains.kotlin</groupId>
+            <artifactId>kotlin-stdlib</artifactId>
+            <version>${kotlin.version}</version>
+        </dependency>
```

Comment:
**suggestion**: I'd prefer to avoid adding this extra dependency for all (also non-Kotlin) users.
It is only used to detect Kotlin classes, which can just as easily be steered through a `JacksonOption` and then be applied to all types equally.

At the same time I was just wondering: since the Jackson library supports those annotations on constructor parameters as well (also in Java classes), there is nothing Kotlin-specific about this functionality then.
It just happens to apply to Kotlin data classes too, sure, but the more generic feature to be added here would be to always consider constructor parameters as well as annotations on the field and its getter.

Reply:
In that case, would you prefer checking for the annotation in the appropriate ctor parameter in `FieldScope.getAnnotationConsideringFieldAndGetter`? We can add an `Option` to enable this extra check